lib/python/bind_operators.h

```diff
@@ -83,6 +83,90 @@ void bind_astype(py::class_<T, Ignored...> &c) {
         :rtype: Union[scipp.Variable, scipp.DataArray])");
 }
 
+template <class T, class... Ignored>
+void bind_to(py::class_<T, Ignored...> &c) {
+  c.def(
+      "to",
+      [](const T &self, const py::object &dt,
+         const std::optional<ProtoUnit> unit, const bool copy) {
+        using scipp::core::dtype;
+
+        const auto src_dtype(self.dtype());
+        const auto dest_dtype(dt.is_none() ? src_dtype : scipp_dtype(dt));
+        const auto dest_units(unit.has_value() ? make_unit(unit.value())
+                                               : self.unit());
+
+        [[maybe_unused]] py::gil_scoped_release release;
+
+        bool convert_dtype_first;
+
+        if (src_dtype == dest_dtype) {
+          return to_unit(self, dest_units,
+                         copy ? scipp::CopyPolicy::Always
+                              : scipp::CopyPolicy::TryAvoid);
+        } else if (dest_dtype == dtype<double>) {
```

Comment:
This can be simplified using
```.cpp
bool is_int(DType tp);
bool is_float(DType tp);
```
from "core/type.h"

Reply:
I think this has to differentiate between the two float types and give preference to `double` if either source or dest is `double`, before looking for either of them being `float`? Unless I've misunderstood what you are suggesting...

Reply:
Ah, I see. I didn't really read them in order.